music21/stream/__init__.py

```diff
@@ -2709,6 +2709,105 @@ def replaceDerived(startSite=self):
 
         replaceDerived()
 
+    def splitAtDurations(self, *, recurse=False) -> Optional[Dict]:
+        '''
+        Overrides base method :meth:`~music21.base.Music21Object.splitAtDurations`
+        so that once each note or rest in the stream having a complex duration is split
+        into similar, shorter elements representing each duration component,
+        the original GeneralNote is actually replaced in the stream where it was found
+        with those new elements.
+
+        Returns a dictionary of the replacements/insertions made, if any, consisting
+        of the stream container acted on (key) and the list of inserted elements (value).
+
+        >>> s = stream.Stream()
+        >>> s.insert(note.Note(quarterLength=5.0))
+        >>> post = s.splitAtDurations()
+        >>> post
+        {<music21.stream.Stream 0x11e1353a0>: [<music21.note.Note C>, <music21.note.Note C>]}
+        >>> [n.duration for n in s]
+        [<music21.duration.Duration 4.0>, <music21.duration.Duration 1.0>]
+
+        Unless `recurse=True`, notes in substreams will not be found.
+
+        >>> s2 = stream.Score()
+        >>> p = stream.Part([note.Note(quarterLength=5)])
+        >>> s2.append(p)
+        >>> s2.splitAtDurations()
+        >>> s2.splitAtDurations(recurse=True)
+        {<music21.stream.Part 0x11486b250>: [<music21.note.Note C>, <music21.note.Note C>]}
+
+        `recurse=True` should not be necessary to find elements in streams
+        without substreams, such as a loose Voice:
+
+        >>> v = stream.Voice([note.Note(quarterLength=5.5)])
+        >>> v.splitAtDurations()
+        {<music21.stream.Voice 0x11fcf8580>: [<music21.note.Note C>, <music21.note.Note C>]}
+
+        But a Voice in a Measure (most common) will not be found without `recurse`:
+
+        >>> m = stream.Measure()
+        >>> v2 = stream.Voice([note.Note(quarterLength=5.25)])
+        >>> m.insert(v2)
+        >>> m.splitAtDurations() is None
+        True
+
+        For any spanner containing the element being removed, the first or last of the
+        replacing components replaces the removed element
+        (according to whether it was first or last in the spanner.)
+
+        >>> s3 = stream.Stream()
+        >>> n1 = note.Note(quarterLength=5)
+        >>> n2 = note.Note(quarterLength=5)
+        >>> s3.append([n1, n2])
+        >>> s3.insert(0, spanner.Slur([n1, n2]))
+        >>> post = s3.splitAtDurations()
+        >>> s3.spanners.first().getFirst() is n1
+        False
+        >>> s3.spanners.first().getFirst().duration
+        <music21.duration.Duration 4.0>
+        >>> s3.spanners.first().getLast().duration
+        <music21.duration.Duration 1.0>
+        '''
+
+        def processContainer(container: Stream):
+            replacements: List[note.GeneralNote] = []
+            for noteObj in container.getElementsByClass('GeneralNote'):
+                if noteObj.duration.type == 'complex':
```

Comment:
I much prefer, `if element.duration.type != 'complex':  continue` to reduce the indentation.